src/tests/testAuthConfig.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef _SQUID__SRC_TESTS_TESTAUTHCONFIG_H
+#define _SQUID__SRC_TESTS_TESTAUTHCONFIG_H
+
+#if USE_AUTH
+
+#include "compat/cppunit.h"
+
+class testAuthConfig : public CPPUNIT_NS::TestFixture
```

Comment:
If possible, please move this helper class into the corresponding .cc file to simplify. Nobody reuses this class now, and future (correct) reuse of this leaf test class is unlikely without substantial API changes.

Reply:
?huh?  This is how unit tests are defined for Squid. The manager class for Auth::Config unit tests gets its own files.

Reply:
I see no good reason to repeat that mistake (i.e. create unnecessary header files), and your response does not give me one.

Reply:
We disagree on that style being "a mistake". It is simply a code style which was chosen decades ago and been used consistently ever since. More consistently than other styles in Squid.

This PR is scoped at fixing a specific bug and adding tests to ensure no future problems can occur due to the fix.

Reply:
I disagree that creating headers for declarations not meant for sharing across translation units should be treated as a style issue. IMO, style excuses should be limited to issues that cannot be decided based on merits/reason/facts. These headers is not such an issue -- they are objectively a noticeable and unnecessary overhead.

Yes, it looks like we have been doing it wrong for many years, but I do not think that tradition alone is a valid excuse for doing it differently _now_. There are numerous problems with old unit tests. We should be doing the right thing when adding new ones, especially when it is trivial to do so.

Reply:
This is a blocker bug for v5. It needs to be back ported, we cannot simply change how unit tests are built and work this late in the life cycle of v5. For *this* PR we have to stick with the old way.

I do have WIP refactoring unit tests that this policy change can be included with. So v6 or so can start the migration, just not here.

Reply:
> It needs to be back ported

As you know, IMO, this unit test should not be added to master/v6 in the first place. If it is added despite my objections, then it does not need to be backported to v5. And if it is backported, then ...

> we cannot simply change how unit tests are built and work this late in the life cycle of v5

... this change request does not have a material effect on backporting AFAICT: I see no reason why the no-header implementation cannot be ported to v5 "as is", just like a with-header implementation would be.

> this policy change

There is no policy change here IMO, just an application of a common sense answer to a basic C++ development question: What code belongs to a header file? If you think that some existing policy forces us to create unneeded C++ headers, then let's work on removing or adjusting that policy.